src/runtime/runtime_data.cs

```diff
@@ -31,6 +31,20 @@ public static Type FormatterType
 
         public static ICLRObjectStorer WrappersStorer { get; set; }
 
+        /// <summary>
+        /// Clears the old "clr_data" entry if a previous one is present.
+        /// </summary>
+        static void ClearCLRData ()
+        {
+            IntPtr capsule = PySys_GetObject("clr_data").DangerousGetAddressUnchecked();
+            if (capsule != IntPtr.Zero)
+            {
+                IntPtr oldData = PyCapsule_GetPointer(capsule, null);
+                PyMem_Free(oldData);
+                PyCapsule_SetPointer(capsule, IntPtr.Zero);
+            }
```

Comment:
Now that you never own the pointer, you can use `BorrowedReference` and never call "Dangerous" methods on it.

Reply:
> Make PyCapsule_GetPointer and PyCapsule_SetPointer take first argument as BorrowedReference

I don't think this is semantically correct. These methods can operate on something other than BorrowedReferences

Reply:
No, this is exactly right. The method borrows a capsule object in a similar way we use borrowed references.

Reply:
In this use case, it is. However in the case where I create the new capsule (and receive a `NewReference`), and then set the capsule's pointer with `PyCapsule_SetPointer`, well.. it's a bit weird. But as I said in another reply, `NewReference` in implicitly castable to borrowed reference, which makes it transparent to the user.